source/server-discovery-and-monitoring/tests/unified/interruptInUse-pool-clear.yml

```diff
@@ -0,0 +1,322 @@
+        object: testRunner
+        arguments:
+          client: *setupClient
+          failPoint:
+            configureFailPoint: failCommand
+            mode:
+              times: 2
+            data:
+              failCommands:
+                - hello
+                - isMaster
+              blockConnection:  true,
+              blockTimeMS: 1500,                
+              appName: interruptInUse        
+      - name: waitForThread
+        object: testRunner,
+        arguments:
+          thread: *thread1
+
+    expectEvents:
+      - client: *client
+        eventType: command
+        events:
+          - commandStartedEvent:
+              commandName: insert
+          - commandStartedEvent:
+              commandName: find
+          - commandFailedEvent:
+              commandName: find
+      - client: *client
+        eventType: cmap
+        events:
+          - connectionCheckedOutEvent: {}
+          - connectionCheckedInEvent: {}
+          - connectionCheckedOutEvent: {}
+          - poolClearedEvent:
+              interruptInUseConnections: true
+          - connectionCheckedInEvent: {}
+          - connectionClosedEvent: {}
+
+    outcome:
+      - collectionName: *collectionName
+        databaseName: *databaseName
+        documents:
+          - _id: 1
+
+  - description: Error returned from connection pool clear with interruptInUseConnections=true is retryable
+    operations:
+      - name: createEntities
+        object: testRunner
+        arguments:
+          entities:
+            - client:
+                id: &client client
+                useMultipleMongoses: false
+                observeEvents:
+                  - poolClearedEvent
+                  - connectionClosedEvent
+                  - commandStartedEvent
+                  - commandFailedEvent
+                  - commandSucceededEvent
+                  - connectionCheckedOutEvent
+                  - connectionCheckedInEvent
+                uriOptions:
+                  connectTimeoutMS: 500
+                  heartbeatFrequencyMS: 500
+                  appname: interruptInUseRetryable
+                  retryReads: true
+                  minPoolSize: 0
+            - database:
+                id: &database database
+                client: *client
+                databaseName: *databaseName
+            - collection:
+                id: &collection collection
+                database: *database
+                collectionName: *collectionName
+            - thread:
+                id: &thread1 thread1
+      - name: insertOne
+        object: *collection
+        arguments:
+          document: { _id: 1 }
+      # simulate a long-running query
+      - name: runOnThread
+        object: testRunner
+        arguments:
+          thread: *thread1
+          operation:
+            name: find
+            object: *collection
+            arguments:
+              filter:
+                $where : sleep(5000) || true
```

Comment:
Could we use `sleep(2000)`  (or less) to make this test faster?

Reply:
done